torch_geometric/datasets/generators/graph_generator.py

```diff
@@ -0,0 +1,57 @@
+from typing import Callable, Optional
+
+import torch
+
+from torch_geometric.data import Data
+
+
+class GraphGenerator:
+    def __init__(self, motif: Optional[Callable], num_nodes: int = 300):
```

Comment:
we could have an option to set the seed here. By default we'll use the same seed, for reproducibility. But if user wants to randomize, they are allowed to set the option and we could generate different graph datasets with the same generator.

Reply:
Thanks for all the comments!
I added a seed call before generating the base graph. I'm not sure if this approach is too naive, but I ran a couple of tests re-using the generator to create other datasets:

```
otif = Motif('house')
generator = BAGraph(num_nodes=300, num_motifs=80, motif=motif, seed=444)
dataset = ExplainerDataset(generator)
data1 = dataset[0]

np.random.rand(3,2)
torch.rand(2, 3)

generator.num_nodes = 300
generator.seed = 1234
dataset = ExplainerDataset(generator)
data2 = dataset[0]

generator.num_nodes = 300
generator.seed = 444 # initial seed
dataset = ExplainerDataset(generator)
data3 = dataset[0]

print("data1", data1.edge_index)
print("data2", data2.edge_index)
print("data3", data3.edge_index)
print(data1.edge_index == data3.edge_index)

Output:

data1 tensor([[  0,   0,   0,  ..., 699,  21, 695],
        [  2,   5,   7,  ..., 696, 695,  21]])
data2 tensor([[  0,   0,   0,  ..., 699, 240, 695],
        [  6,  10,  15,  ..., 696, 695, 240]])
data3 tensor([[  0,   0,   0,  ..., 699,  21, 695],
        [  2,   5,   7,  ..., 696, 695,  21]])
tensor([[True, True, True,  ..., True, True, True],
        [True, True, True,  ..., True, True, True]])

```